### PR TITLE
chore(Tabs): add tests for overflow tab behavior

### LINKED ISSUE #7773: Tabs - add tests for horizontal overflow tabs
Add tests for the horizontal overflow tabs behavior of the Tabs component added in #7742

### PR DESCRIPTION
**Pull request by wise-king-sullyman**
**What**: Closes #7773 

**Additional issues**:

### CODE REVIEW

File: packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx [hunk 1]
```diff
@@ -0,0 +1,393 @@
...
+        ]}
+      />
+    </TabsContext.Provider>
+  );
+
+  expect(screen.getByRole('tab')).toHaveTextContent('Tab one');
+});
+
+test('Uses custom overflow tab title text provided via defaultTitleText', () => {
+  render(<OverflowTab defaultTitleText="Test" />);
```
Comment (jeffpuzzo) [inline packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx:278]: Should this prop actually be called `titleText` instead? Since we're passing a "custom" value to something called "default".

File: packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx [hunk 2]
```diff
@@ -0,0 +1,393 @@
...
+
+  userEvent.click(overflowTab);
+
+  expect(overflowTab).toHaveAttribute('aria-expanded', 'true');
+});
+
+test('Passes Popper popperMatchesTriggerWidth set to false', () => {
+  render(<OverflowTab />);
+
+  expect(screen.getByText('Popper matches trigger width: false')).toBeVisible();
```
Comment (jeffpuzzo) [inline packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx:335]: nit: I might reference the mocked Popper above here in some way maybe via comment since it's so far away from that mock and it might not be obvious jumping to this test where the "Popper matches trigger width" text comes from. Same with the tests below.

File: packages/react-integration/demo-app-ts/src/components/demos/TabsDemo/TabsHorizontalOverflowDemo.tsx [hunk 3]
```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { Tabs, Tab, TabTitleText, Checkbox } from '@patternfly/react-core';
+
+export const TabsHorizontalOverflowDemo: React.FunctionComponent = () => {
+  const [activeTabKey, setActiveTabKey] = React.useState<string | number>(0);
+  const [showTabCount, setShowTabCount] = React.useState(false);
+  const [defaultTitleText, setDefaultTitleText] = React.useState(undefined);
+  const [toggleAriaLabel, setToggleAriaLabel] = React.useState(undefined);
+
+  const handleTabClick = (_event: any, tabIndex: string | number) => {
```
Comment (jeffpuzzo) [inline packages/react-integration/demo-app-ts/src/components/demos/TabsDemo/TabsHorizontalOverflowDemo.tsx:10]: nit: Is this event meant to be any? I understand it's not used, but maybe still we could use to correct type.

File: packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx (hunk 1 above)
Comment (jeffpuzzo) [reply packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx:278]: Spoke about this in slack, I see how the naming is justified to avoid the possible implication that a selected overflow item's name won't take precedence over this `defaultTitleText`. We might also consider `initial` as a keyword, but I'm ok with what it is.

**Review by jeffpuzzo — APPROVED** [review]
This looks great! All remaining comments are nitpicks and non-blocking.

File: packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+    </TabsContext.Provider>
+  );
+
+  userEvent.click(screen.getByRole('tab'));
+
+  // This assertion relies on the structure of the Popper mock to verify the correct props are being sent to Popper
+  expect(screen.getByText('Selected: true')).toBeVisible();
+});
+
+test('Matches snapshot', () => {
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx:386]: Nit:

```suggestion
test('Matches snapshot when expanded', () => {
```

Just for clarifying the state of the component in the snapshot, but not a blocker.

File: packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+        {props.children}
+        <button onClick={() => props.onSelect('mouseEvent', 1, 'fakeRef')}>Select</button>
+      </div>
+    )),
+    MenuContent: ({ children }) => <div data-testid="menu-content-mock">{children}</div>,
+    MenuList: ({ children }) => <ul data-testid="menu-list-mock">{children}</ul>,
+    MenuItem: ({ children, itemId, isSelected }) => (
+      <>
+        <li id={`${itemId}-menu-item-mock`}>{children}</li>
+        <p>Selected: {`${isSelected}`}</p>
```
Comment (thatblindgeye) [inline packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx:33]: Nit:
```suggestion
        <li id={`${itemId}-menu-item-mock`}>
          {children}
          <p>Selected: {`${isSelected}`}</p>
        </li>
```
Or something along these lines, just to have valid HTML in the test. Overall may not be a huge thing since it's a test so not blocking over it, but let me know what you think in regards to catching this sort of stuff in future tests.

**Review by thatblindgeye — APPROVED** [review]
This looks great! Just had a couple of non-blocking nitpick below, but otherwise 🔥

**Review by tlabaj — APPROVED** [review]

File: packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx (hunk 5 above)
Comment (wise-king-sullyman) [reply packages/react-core/src/components/Tabs/__tests__/OverflowTab.test.tsx:33]: I think that's a great suggestion 🙂
